scripts/ci_aws_bootstrap.sh

```diff
@@ -2,76 +2,90 @@
 
 set -euo pipefail
 
-if [[ -z "${RUN_ALL:-}" ]]; then
-    RUN_ALL=0
-fi
+RUN_ALL=${RUN_ALL:-0}
+export VLLM_BUILDKITE_BRANCH="pipeline_gen"
+
+generate_pipeline() {
+    python -m pip install click pydantic
+    
+    # Download necessary files
+    for file in pipeline_generator.py plugin.py step.py utils.py; do
+        curl -o .buildkite/$file https://raw.githubusercontent.com/vllm-project/buildkite-ci/$VLLM_BUILDKITE_BRANCH/scripts/pipeline_generator/$file
+    done
+    
+    # Generate and upload pipeline
+    python .buildkite/pipeline_generator.py --run_all=$RUN_ALL --list_file_diff="$LIST_FILE_DIFF"
+    cat .buildkite/pipeline.yaml
+    buildkite-agent pipeline upload .buildkite/pipeline.yaml
+    exit 0
+}
 
 upload_pipeline() {
     echo "Uploading pipeline..."
     ls .buildkite || buildkite-agent annotate --style error 'Please merge upstream main branch for buildkite CI'
+    
+    # Install minijinja-cli
     curl -sSfL https://github.com/mitsuhiko/minijinja/releases/latest/download/minijinja-cli-installer.sh | sh
     source /var/lib/buildkite-agent/.cargo/env
+    
     if [ $BUILDKITE_PIPELINE_SLUG == "fastcheck" ]; then
-        if [ ! -e ".buildkite/test-template-fastcheck.j2" ]; then
-            curl -o .buildkite/test-template-fastcheck.j2 https://raw.githubusercontent.com/vllm-project/buildkite-ci/main/scripts/test-template-fastcheck.j2
-        fi
-        cd .buildkite && minijinja-cli test-template-fastcheck.j2 test-pipeline.yaml > pipeline.yml
-        cat pipeline.yml
-        buildkite-agent pipeline upload pipeline.yml
-        exit 0
+        handle_fastcheck
+    else
+        handle_regular_pipeline
     fi
-    if [ ! -e ".buildkite/test-template.j2" ]; then
+}
+
+handle_fastcheck() {
+    [ ! -e ".buildkite/test-template-fastcheck.j2" ] && \
+        curl -o .buildkite/test-template-fastcheck.j2 https://raw.githubusercontent.com/vllm-project/buildkite-ci/main/scripts/test-template-fastcheck.j2
+    
+    cd .buildkite && minijinja-cli test-template-fastcheck.j2 test-pipeline.yaml > pipeline.yml
```

Comment:
this will change the working directory, is that intended?

use `tee` rather than a separate cat? it will allow you to also see the output when the previous line has error.